dashboard/app/models/user.rb

```diff
@@ -349,7 +349,11 @@ def self.find_channel_owner(encrypted_channel_id)
     User.find(user_id)
   end
 
-  validate :presence_of_email, if: :teacher?
+  # On creation, always check for an email for teachers. Otherwise, only check if
+  # it has changed, so that we can still update other parts of the model if the
+  # teacher has no email, which is true of some old accounts
+  validate :presence_of_email, if: 'teacher? && email_changed?'
+  validate :presence_of_email, if: :teacher?, on: :create
```

Comment:
Not sure. Presumably, you could test locally by seeing if you can create a teacher without an  empty email address?